infra/label_evaluator_step_functions/handlers_traced/discover_patterns_traced.py

```
@@ -0,0 +1,469 @@
+def load_json_from_s3(bucket: str, key: str) -> dict[str, Any] | None:
+    """Load JSON data from S3, return None if not found."""
+    try:
+        response = s3.get_object(Bucket=bucket, Key=key)
+        return json.loads(response["Body"].read().decode("utf-8"))
+    except ClientError as e:
+        error_code = e.response.get("Error", {}).get("Code")
+        if error_code in {"NoSuchKey", "404"}:
+            return None
+        logger.warning("Error loading %s: %s", key, e)
+        raise
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)